### ISSUE TITLE
`:exe "nmap x \<F1>"` makes both maparg() and nvim_get_keymap() report invalid data

### ISSUE THREAD

**Issue by ZyX-I** [neovim/neovim#6953]
- `nvim --version`: 35898cff5d1d6dc60e0d7b87bfe106539453b031
- Vim (version: 8.0.692) behaves differently? See vim/vim#1816.

### Steps to reproduce

```
nvim --headless -u NONE -i NONE -N --cmd 'execute "nmap x \<F1>"' --cmd 'echo nvim_get_keymap("n")|echo maparg("x")|echo maparg("x", "n", 0, 1)' --cmd cq
```

### Expected behaviour

All show something like `<Char-0x80>k1` in rhs. Or, at least, `<M-C-@>k1` like Vim does.

### Actual behaviour

```
[{'expr': 0, 'noremap': 0, 'lhs': 'x', 'mode': 'n', 'nowait': 0, 'silent': 0, 'sid': -2, 'rhs': 'Xk1', 'buffer': 0}]
Xk1
{'silent': 0, 'noremap': 0, 'lhs': 'x', 'mode': 'n', 'nowait': 0, 'expr': 0, 'sid': -2, 'rhs': '~@<80>k1', 'buffer': 0}
```

Note that second maparg() call outputs something even more strange: `~@<80>k1`. The bad thing here may be yet another bug: literally it is `\x80k1`, as expected. But `\x80` is output as `~@<80>` while `strtrans("\x80")` is only `<80>`. In all other cases `Xk1` is literal `Xk1`.

**Comment by ZyX-I** [comment]
The problem is only partially seen on e333957a1a9ae64b7daa36e08fd1df583114d4ba: due to the changes made bug seen with `maparg('x')` moved to `nvim_get_keymap()` as well. (I.e. before my PR both 'rhs' were `~@<80>k1`, but `Xk1` is the same before and after it.)

**Comment by ZyX-I** [comment]
Xk1 may also be seen on v0.2.0, as well as `~@<80>`.

**Comment by ZyX-I** [comment]
On v0.1.0 I see Xk1, but I do not see `~@`, only `<80>`.

**Comment by ZyX-I** [comment]
The earliest revision which still compiles (ec5e94518) still shows Xk1 in `maparg('x')` output.

**Comment by zeertzjq** [comment]
Fixed by #18491
